Packs/SplunkPyPreRelease/Integrations/SplunkPyPreRelease/SplunkPyPreRelease.py

```diff
@@ -307,30 +310,25 @@ def fetch_notables(service, cache_object=None, enrich_notables=False):
             # want to add data to the integration context (which will ruin the logic of the cache object)
             last_run_data.update({DUMMY: DUMMY})
 
-    if len(incidents) == 0:
+    if len(incidents) == 0 and len(reader) < batch_size:
         next_run = get_next_start_time(last_run_time, now, False)
-        extensive_log('[SplunkPyPreRelease] SplunkPyPreRelease - Next run time with no incidents found: {}'.format(next_run))
+        extensive_log('[SplunkPyPreRelease] SplunkPyPreRelease - Next run time with no incidents found: {}.'.format(next_run))
         new_last_run = {
             'time': next_run,
-            'offset': 0,
-            'found_incidents_ids': last_run_fetched_ids
-        }
-    elif len(incidents) < FETCH_LIMIT:
-        latest_incident_fetched_time = get_latest_incident_time(incidents)
-        next_run = get_next_start_time(latest_incident_fetched_time, now)
-        extensive_log('[SplunkPyPreRelease] SplunkPyPreRelease - Next run time with some incidents found: {}'.format(next_run))
-        new_last_run = {
-            'time': next_run,
-            'offset': 0,
-            'found_incidents_ids': last_run_fetched_ids
+            'found_incidents_ids': last_run_fetched_ids,
+            'batch_size': default_batch_size
         }
     else:
+        if len(reader) == batch_size:
+            batch_size += default_batch_size
+        latest_incident_fetched_time = get_latest_incident_time(incidents)
+        next_run = get_next_start_time(latest_incident_fetched_time, now, were_new_incidents_found=True)
```

Comment:
There may be no new incidents.
We need to change the logic here to make sure we handle the cases differently

Reply:
handled in the last commit